pytorch_lightning/trainer/connectors/checkpoint_connector.py

```diff
@@ -33,7 +34,7 @@
 
 
 class CheckpointConnector:
-    def __init__(self, trainer, resume_from_checkpoint: Optional[Union[str, Path]] = None):
+    def __init__(self, trainer: "pl.Trainer", resume_from_checkpoint: Optional[Union[str, Path]] = None) -> None:
```

Comment:
let's use this everywhere for the path: https://github.com/PyTorchLightning/pytorch-lightning/blob/089ae9b3e82ddc31942e315294e31e48c0a899db/pytorch_lightning/utilities/types.py#L35